model/src/pyrenew/process/rtperiodicdiff.py

```diff
@@ -207,10 +209,11 @@ def sample(
 
         # Expanding according to the number of days
         if duration is None:
-            duration = self.n_obs
-        elif duration > self.n_obs:
+            duration = self.n_timepoints
+        elif duration > self.n_timepoints:
             raise ValueError(
-                f"Duration should be less than or equal to n_obs ({self.n_obs})."
+                "Duration should be less than or equal "
+                f"to n_timepoints ({self.n_timepoints })."
```

Comment:
It appears that there's both a `self.n_timepoints` class attributes and also a `duration` passed as a mandatory argument to `self.sample()`. Should that definitely be the case?

Reply:
When I implemented this, I had in mind an error I saw about `jit` and objects sizes. When sizing/filtering as a function of a sampled quantity, jax.jit was not happy. This is not the case, so I removed `n_timepoints` in favor of `duration` in 4e4a12e.